### ISSUE TITLE
freebayes-parrallel report snp but wih multple bases

### ISSUE THREAD

**Issue by grayshiwb512** [freebayes/freebayes#161]
Dear Erik,
Hope everything running well for you.
Thanks for your time and efforts in freebayes development.
Recently, we had a strange observation that I’d like you to help look into.
When we using your developed freebayes-Parrallel script to do the variant calling, we have following observation,

Chr-05  27410196        snp     G       A       10.1    311
Chr-05  27410208        snp     TCA     TCC     58.1    307
Chr-05  27410212        snp     G       A       8.4     318
Chr-05  27427879        snp     C       T       4922.0  330
Chr-05  27428451        snp     CAGGACTG        CAGGACTA        1725.8  329
Chr-05  27428466        snp     C       G       1.5     359

Here you can see, some position the variant caller returned the snp but with multiple bases.
Do you have idea why this will happen?
Looking forward to hearing from you.
Thanks in advance
Paul

**Comment by tseemann** [comment]
@grayshiwb512  Does this occur when you don't use ```freebayes-parallel``` ?

**Comment by noahrose** [comment]
Hi there,

I'm seeing this behavior too, without freebayes parallel, though I am using a targets file. Here's an example:

LISTak.vcf:contig600169	643	.	CTGGC	TTGGC	968.707	.	AB=0.35443;ABP=46.6322;AC=10;AF=0.135135;AN=74;AO=96;CIGAR=1X4M;DP=945;DPB=1047.4;DPRA=0.936047;EPP=39.2015;EPPR=155.203;GTI=4;LEN=1;MEANALT=1.5;MQM=39.6667;MQMR=41.5043;NS=37;NUMALT=1;ODDS=0.577152;PAIRED=0;PAIREDR=0;PAO=84.5;PQA=2555.5;PQR=2626;PRO=92;QA=3308;QR=28368;RO=815;RPL=24;RPP=55.1256;RPPR=10.4946;RPR=72;RUN=1;SAF=8;SAP=147.775;SAR=88;SRF=481;SRP=60.585;SRR=334;TYPE=snp

 Vcf check seems happy with the files, and the SNP isn't at the region boundary (here the region goes from 1 to 866, and the SNP is at 643). If I visually inspect the fasta, the ref sequence is all there as expected. Have you had any figuring this one out? Thanks in advance for any help!

Noah

**Comment by noahrose** [comment]
Hi there,

A quick update: this behavior (with a SNP being returned with multiple padding bases) seems to happen when a SNP is returned with only a single alternate allele in the VCF, but then the NUMALT field is set to 2. Like this:

NUMALT=2.... GCAC	GCAT

I think what is happening is that two nearby SNPs are detected at a locus, they get lumped into a single variant, and then one gets dropped out of reporting or something. In fact, sometimes when two nearby SNPs are totally unlinked, they get returned as one variant:

NUMALT=2 ... AAAG	AAAA,TAAG

It seems to me that this is not desirable behavior, since the SNPs are not linked into a true mnp, and have in fact been labeled a SNP, which I would think should always have just one base. I would think these two SNPs should be reported as two different single base loci. Let me know if I have misunderstood the desired effect or if more information would be helpful. This is with standard freebayes (not parallel).

**Comment by ekg** [comment]
Nice catch on the numalt field. The documentation in the VCF header should
clarify that this is the number of alternate alleles considered by the
algorithm at the locus. We can introduce an additional field to specify the
number of alleles, but this is trivially available in the record (the
number of ALTs).

Now, I am not going to change the behavior in the second case. There are
two problems. Firstly, freebayes has attempted a haplotype call for reasons
other than perfect linkage of SNPs. One example is a candidate deletion
overlapping the site. Another is low complexity sequence, where we have to
make haplotype calls to drastically reduce alignment artifact driven calls.
Secondly, it may be that a candidate allele had the two SNPs in linkage.

In summary, I want to let the algorithm explain what it is doing and leave
the normalization of variants up to downstream programs. Does this clarify
what you are seeing?

**Comment by noahrose** [comment]
That makes sense, and I totally appreciate the algorithmic transparency. Additional documentation of exactly when the vcf reports algorithmic properties that may differ from a direct interpretation of the output would definitely be useful and save some head scratching. Thanks the support and for all of your work on freebayes, I appreciate it!

N

**Comment by liggettla** [comment]
I am also receiving this type of result and I'm a bit unclear if there is a way to suppress this or not. The difficulty for me is that when a I get these multi-base variants, the variant locus changes like the following between different vcf files:

1.vcf
chr1    100    CTGGC    CTGGT

2.vcf
chr1    102    GGC    GGT

3.vcf
chr1    104    C    T

This makes it challenging to automatically parse through a large number of .vcf files and identify repeating variants by locus. Is there some way to force just the single variant output?

Sorry if I am missing the answer from above, and thanks for all the work on this project.

**Comment by AlistairNWard** [comment]
You should try post-processing the vcf output using vt normalize (
https://github.com/atks/vt). This will generate left-aligned and
parsimonious representation of alleles. After vt normalize, the three vcf
files above, would contain the representation in 3.vcf.


Alistair Ward, PhD
Director, Research and Science
Department of Human Genetics
USTAR Center for Genetic Discovery
Eccles Institute of Human Genetics
University of Utah School of Medicine​
15 North 2030 East, Room 7140
Salt Lake City, UT 84112-5330

Co-founder, Frameshift labs, LLC

**Comment by ekg** [comment]
I think you also want to use vt decompose_blocksub. There is also
vcfallelicprimitives in vcflib which does the same thing.

**Comment by liggettla** [comment]
I'm having trouble getting vt normalize (v0.5) to work so I haven't checked that output. But vt decompose_blocksub seems to entirely take care of the problem. All the variants are left aligned and parsimonious in the output. I appreciate the help.

**Comment by atks** [comment]
@liggettla - out of curiosity, may I know what was wrong with the output from vt normalize?
